src/lib.rs

```diff
@@ -813,16 +806,17 @@ mod tests {
         case::normal_le(0xDDCC_BBAA, Endian::Little, None, vec![0xAA, 0xBB, 0xCC, 0xDD]),
         case::normal_be(0xDDCC_BBAA, Endian::Big, None, vec![0xDD, 0xCC, 0xBB, 0xAA]),
         case::bit_size_le_smaller(0x03AB, Endian::Little, Some(10), vec![0xAB, 0b11_000000]),
-        case::bit_size_be_smaller(0x03AB, Endian::Big, Some(10), vec![0b11, 0xAB]),
+        case::bit_size_be_smaller(0x03AB, Endian::Big, Some(10), vec![0b11_1010_10, 0b11_000000]),
```

Comment:
I had to change the layout of this output vec because of the alignment of the bits written to the output `BitVec`. Originally, the `BitVec` returned by `DekuWrite::write` was created directly from the array of bytes created from an integer (in this case, `0x03AB.to_be_bytes()  == [0b000000_11, 0b1010_1011]`), with an offset pointing to the first byte of the output.

With the new API, an empty `BitVec` is created by the caller and passed in, where the bytes are written MSB-first, so that the backing array is `[0b11_1010_10, 0b11_000000]` - the previous array, but shifted left by 6 bits.

When `BitVec::into_vec` is called in the test harness, it reuses the backing array without shifting or re-aligning the data it contains, so the offset to the first bit is unchanged. Originally, this meant 6 bits of left-padding, which is now gone.